core-plugins/src/main/java/io/cdap/plugin/batch/action/EmailAction.java

```diff
@@ -107,37 +109,39 @@ public Config() {
       includeWorkflowToken = false;
     }
 
-    public void validate() {
-      super.validate();
+    public void validate(FailureCollector collector) {
+      super.validate(collector);
 
       if (!containsMacro("username") && (Strings.isNullOrEmpty(username) ^ Strings.isNullOrEmpty(password))) {
-        throw new IllegalArgumentException("You must either set both username and password or " +
-                                             "neither username nor password.");
+        collector.addFailure("You must either set both username and password or neither username nor password.",
+                  "Leave username and password fields empty or provide values for both fields")
```

Comment:
fix indentation to match the rest of the project

Reply:
We don't always need to set a corrective action. When it is basically just restating the error message like here, it's better to leave it out.

Reply:
Removed corrective action and reformatted file